server/sonar-server/src/main/java/org/sonar/server/qualityprofile/RuleActivator.java

```diff
@@ -60,79 +62,65 @@
   private final System2 system2;
   private final DbClient db;
   private final TypeValidations typeValidations;
-  private final RuleActivatorContextFactory contextFactory;
-  private final RuleIndex ruleIndex;
-  private final ActiveRuleIndexer activeRuleIndexer;
   private final UserSession userSession;
 
-  public RuleActivator(System2 system2, DbClient db, RuleIndex ruleIndex,
-    RuleActivatorContextFactory contextFactory, TypeValidations typeValidations,
-    ActiveRuleIndexer activeRuleIndexer, UserSession userSession) {
+  public RuleActivator(System2 system2, DbClient db, TypeValidations typeValidations, UserSession userSession) {
     this.system2 = system2;
     this.db = db;
-    this.ruleIndex = ruleIndex;
-    this.contextFactory = contextFactory;
     this.typeValidations = typeValidations;
-    this.activeRuleIndexer = activeRuleIndexer;
     this.userSession = userSession;
   }
 
-  public List<ActiveRuleChange> activateOnBuiltInRulesProfile(DbSession dbSession, RuleActivation activation, RulesProfileDto rulesProfile) {
-    checkArgument(rulesProfile.isBuiltIn(), "Rules profile must be a built-in profile: " + rulesProfile.getKee());
-    RuleActivatorContext context = contextFactory.createForBuiltIn(dbSession, activation.getRuleKey(), rulesProfile);
+  public List<ActiveRuleChange> activate(DbSession dbSession, RuleActivation activation, RuleActivationContext context) {
+    context.reset(activation.getRuleKey());
     return doActivate(dbSession, activation, context);
   }
 
-  public List<ActiveRuleChange> activateAndCommit(DbSession dbSession, RuleActivation activation, QProfileDto profile) {
-    List<ActiveRuleChange> changes = activate(dbSession, activation, profile);
-    activeRuleIndexer.commitAndIndex(dbSession, changes);
-    return changes;
-  }
+  private List<ActiveRuleChange> doActivate(DbSession dbSession, RuleActivation activation, RuleActivationContext context) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Refactor this method to reduce its Cognitive Complexity from 30 to the 15 allowed. ([squid:S3776](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3776))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fsb%2FSONAR-10052%2Fqprofiles&issues=AWE-qvP5mckxG3vIth78&open=AWE-qvP5mckxG3vIth78)

[ISSUE_KEY]: AWE-qvP5mckxG3vIth78